### PR TITLE
Feature/add notes

### PR DESCRIPTION
**Pull request by ltwheeler**
Adds a way to store and view notes over time on both a course and instructor.

Adds two input boxes in the assignments view (one for each pivot).
Shows up as a label in print view.

Issue:
https://trello.com/c/9q98Keux/1779-courses-add-a-way-to-store-and-view-notes-about-a-course-or-instructor-over-time

screen shot 2018-07-31 at 7 48 41 am
screen shot 2018-07-31 at 7 48 53 am

### CODE REVIEW

File: app/assignment/directives/instructorAssignmentTable.js [hunk 1]
```diff
@@ -540,6 +562,34 @@ let instructorAssignmentTable = function ($rootScope, AssignmentActionCreators,
 
 			// end on event 'assignmentStateChanged'
 
+			// Handle input box edits
+			element.on("keydown", function(e) {
+				let $el = $(e.target);
+				if ($el.data('event-type') != 'setInstructorNote') { return; }
+
+				if (e.key == 	"Enter") {
```
Comment (cthielen) [inline app/assignment/directives/instructorAssignmentTable.js]: Fix spacing after ==

File: app/assignment/directives/instructorAssignmentTable.js [hunk 2]
```diff
@@ -540,6 +562,34 @@ let instructorAssignmentTable = function ($rootScope, AssignmentActionCreators,
...
+					var scheduleId = $el.data('schedule-id');
+
+					var note = e.target.value;
+
+					AssignmentActionCreators.createOrUpdateInstructorNote(scheduleId, instructorId, note);
+				}
+			});
+
+			// Handle input box edits
+			element.on("change", function(e) {
```
Comment (cthielen) [inline app/assignment/directives/instructorAssignmentTable.js]: Does 'change' handle blur? That seems like a logical time to save.

File: app/assignment/services/assignmentActionCreators.js [hunk 3]
```diff
@@ -34,6 +34,45 @@ class AssignmentActionCreators {
...
+
+				instructorNote = instructorNote ? instructorNote : {
+					scheduleId: scheduleId,
+					instructorId: instructorId
+				};
+
+				instructorNote.note = note;
+
+				_self.AssignmentService.updateInstructorNote(instructorNote).then(function (newInstructorNote) {
+					_self.$rootScope.$emit('toast', { message: "Updated course note", type: "SUCCESS" });
```
Comment (cthielen) [inline app/assignment/services/assignmentActionCreators.js]: Should read "Updated instructor note"

File: app/assignment/services/assignmentActionCreators.js [hunk 4]
```diff
@@ -34,6 +34,45 @@ class AssignmentActionCreators {
...
+					var action = {
+						type: ActionTypes.UPDATE_INSTRUCTOR_NOTE,
+						payload: {
+							instructorNote: newInstructorNote
+						}
+					};
+					_self.AssignmentStateService.reduce(action);
+				}, function (err) {
+					_self.$rootScope.$emit('toast', { message: "Could not update course note.", type: "ERROR" });
```
Comment (cthielen) [inline app/assignment/services/assignmentActionCreators.js]: Should read 'Could not update instructor note'

File: app/assignment/services/assignmentService.js [hunk 5]
```diff
@@ -26,6 +26,12 @@ class AssignmentService {
 	
 				return deferred.promise;
 			},
+			updateCourse: function (course) {
+				return ApiService.put("/api/courseView/courses/" + course.id, course);
+			},
+			updateInstructorNote: function (instructorNote) {
+				return ApiService.put("/api/schedules/" + instructorNote.scheduleId + "/instructors/" + instructorNote.instructorId + "/instructorNotes", instructorNote);
```
Comment (cthielen) [inline app/assignment/services/assignmentService.js]: By this API route, are we storing notes per instructor and per schedule?

File: app/assignment/directives/instructorAssignmentTable.js (hunk 2 above)
Comment (ltwheeler) [reply app/assignment/directives/instructorAssignmentTable.js]: As I understand it "change" is similar to "blur", but only fires if the value was actually changed, whereas blur would fire regardless.
https://api.jquery.com/change/

File: app/assignment/services/assignmentService.js (hunk 5 above)
Comment (ltwheeler) [reply app/assignment/services/assignmentService.js]: That is correct.

**Comment by ltwheeler** [comment]
- Fixed private staff comment modal.
- Updated styles to be taller and use all available space.
- Tweaked border style and added a dividing line.
screen shot 2018-08-10 at 7 44 32 am

**Comment by cthielen** [comment]
Two issues:

Modal instructor notes 'save' button does not close dialog. (Not sure what it did before but it should close on save.)

and

Instructor unavailability modal does not open. JS error:

[Error] TypeError: scope.openUnavailabilityModal is not a function. (In 'scope.openUnavailabilityModal(instructorId)', 'scope.openUnavailabilityModal' is undefined)
	(anonymous function) (instructorAssignmentTable.js:665)
	dispatch (lib.js:5087)

File: app/assignment/css/assignments.css [hunk 6]
```diff
@@ -972,3 +972,26 @@ li.active-term .dropdown-term-row {
...
+	height: 30px;
+}
+
+.add-note__text-area {
+	resize: none;
+	border-color: #e2e2e2;
+}
+
+.assignment__divider {
+  height: 1px;
```
Comment (cthielen) [inline app/assignment/css/assignments.css]: Inconsistent tabbing

File: app/assignment/directives/courseAssignmentTable.js [hunk 7]
```diff
@@ -322,6 +331,30 @@ let courseAssignmentTable = function ($rootScope, AssignmentActionCreators) {
 				}
 			}); // end on event 'assignmentStateChanged'
 
+			// Handle input box edits
+			element.on("keydown", function(e) {
+				let $el = $(e.target);
+				if ($el.data('event-type') != 'setCourseNote') { return; }
+
+				if (e.key == 	"Enter") {
```
Comment (cthielen) [inline app/assignment/directives/courseAssignmentTable.js]: Needless extra spaces after ==

File: app/assignment/directives/instructorAssignmentTable.js (hunk 2 above)
Comment (cthielen) [reply app/assignment/directives/instructorAssignmentTable.js]: Cool, thanks!

**Pull request merged**